Stripping whitespace in <animate> values attribute should use stripLeadingAndTrailingMatchedCharacters(isHTMLSpace<UChar>)

nt1m opened the pull request:
#### e41bfba750f7297186d9401b7af414de5d1d1b58
<pre>
Stripping whitespace in &lt;animate&gt; values attribute should use stripLeadingAndTrailingMatchedCharacters(isHTMLSpace&lt;UChar&gt;)
<a href="https://bugs.webkit.org/show_bug.cgi?id=241867">https://bugs.webkit.org/show_bug.cgi?id=241867</a>

Reviewed by Chris Dumez and Darin Adler.

* Source/WebCore/svg/SVGAnimationElement.cpp:
(WebCore::SVGAnimationElement::attributeContainsJavaScriptURL const):
(WebCore::SVGAnimationElement::parseAttribute):

Canonical link: <a href="https://commits.webkit.org/251835@main">https://commits.webkit.org/251835@main</a>
</pre>

cdumez commented on Source/WebCore/svg/SVGAnimationElement.cpp:
```diff
@@ -155,7 +155,7 @@ bool SVGAnimationElement::attributeContainsJavaScriptURL(const Attribute& attrib
     if (attribute.name() == SVGNames::valuesAttr) {
         for (auto innerValue : StringView(attribute.value()).split(';')) {
-            if (WTF::protocolIsJavaScript(innerValue.stripWhiteSpace()))
+            if (WTF::protocolIsJavaScript(innerValue.stripLeadingAndTrailingMatchedCharacters(isHTMLSpace<UChar>)))
```
Why?

StringView::stripWhiteSpace() is already an alias for StringView::stripLeadingAndTrailingMatchedCharacters(isHTMLSpace<UChar>) AFAIK. The new code is less concise and should have identical behavior.

I may be missing something but then again, the change log doesn't explain anything.

cdumez replied:
Oh:

```
StringView StringView::stripWhiteSpace() const
{
    return stripLeadingAndTrailingMatchedCharacters(isASCIISpace<UChar>);
}
```

So stripWhiteSpace() strips ASCII spaces and your new patch only strips HTML spaces.

cdumez reviewed (approved):
OK. That said, I am not 100% sure that we're meant to use HTML spaces. I found this in the SMIL spec (which the SVG spec seems to reference):
```
S              ::= (#x20 | #x9 | #xD | #xA)+ /* from [[XML11]](https://www.w3.org/TR/SMIL3/smil30.html#smil-refs-ref-XML11) */
```

nt1m committed:
Stripping whitespace in <animate> values attribute should use stripLeadingAndTrailingMatchedCharacters(isHTMLSpace<UChar>)
https://bugs.webkit.org/show_bug.cgi?id=241867

Reviewed by Chris Dumez and Darin Adler.

* Source/WebCore/svg/SVGAnimationElement.cpp:
(WebCore::SVGAnimationElement::attributeContainsJavaScriptURL const):
(WebCore::SVGAnimationElement::parseAttribute):

Canonical link: https://commits.webkit.org/251835@main
